src/models/Preset.ts

```diff
@@ -21,15 +21,21 @@ export interface PropertyArgs {
   property: CommonModel;
 }
 
+export interface AdditionalPropertiesArgs {
+  parentModel: CommonModel;
+  additionalProperties: CommonModel;
+}
 export interface ClassPreset<R extends AbstractRenderer, O extends object = any> extends CommonPreset<R, O> {
   ctor?: (args: PresetArgs<R, O>) => Promise<string> | string;
   property?: (args: PresetArgs<R, O> & PropertyArgs) => Promise<string> | string;
+  additionalProperties?: (args: PresetArgs<R, O> & AdditionalPropertiesArgs) => Promise<string> | string;
```

Comment:
`additionalProperties` should have only `PresetArgs` arguments, why? `parentModel ` is not needed - it's my fault and in next PR I will remove it - we have `model` property for that (it is the same 😅). Also `additionalProperties` field is hardcoded for `additionalProperties` inferred from `model` so you have also possibility to use it by `model`. `additionalProperties` preset should have also possibility to render `patternProperties` etc, so hardcoded for only one field is very bad idea.

```ts
additionalProperties?: (args: PresetArgs<R, O>) => Promise<string> | string;
```